### PR TITLE
Use group resource in migrate reporting

### PR DESCRIPTION
**Pull request by enj**
This change updates the reporting during migration to use the group resource instead of just the resource.

Thus the confusing messages with "the server does not allow this method on the requested resource" for pod metrics will go from:

-n hasha-pro1 pods/nodejs-ex-2-kmzrv

to:

-n hasha-pro1 pods.metrics.k8s.io/nodejs-ex-2-kmzrv

This makes it clear what resource is being referenced.

Bug 1631517

Signed-off-by: Monis Khan <mkhan@redhat.com>

### CODE REVIEW

File: pkg/oc/cli/admin/migrate/migrator.go [hunk 1]
```diff
@@ -594,12 +594,14 @@ type migrateTracker struct {
 func (t *migrateTracker) report(prefix string, info *resource.Info, err error) {
 	ns := info.Namespace
 	if len(ns) > 0 {
-		ns = "-n " + ns
+		ns = " -n " + ns
 	}
+	groupResource := info.Mapping.Resource.GroupResource()
+	groupResourceStr := (&groupResource).String()
```
Comment (juanvallejo) [inline pkg/oc/cli/admin/migrate/migrator.go]: is the string method only defined on a pointer receiver?

File: pkg/oc/cli/admin/migrate/migrator.go [hunk 2]
```diff
@@ -594,12 +594,14 @@ type migrateTracker struct {
...
 	if err != nil {
-		fmt.Fprintf(t.out, "E%s %-10s %s %s/%s: %v\n", timeStampNow(), prefix, ns, info.Mapping.Resource.Resource, info.Name, err)
+		fmt.Fprintf(t.out, "E%s %-10s%s %s/%s: %v\n", timeStampNow(), prefix, ns, groupResourceStr, info.Name, err)
 	} else {
-		fmt.Fprintf(t.out, "I%s %-10s %s %s/%s\n", timeStampNow(), prefix, ns, info.Mapping.Resource.Resource, info.Name)
+		fmt.Fprintf(t.out, "I%s %-10s%s %s/%s\n", timeStampNow(), prefix, ns, groupResourceStr, info.Name)
```
Comment (juanvallejo) [inline pkg/oc/cli/admin/migrate/migrator.go]: why did you remove the space here?

File: pkg/oc/cli/admin/migrate/migrator.go [hunk 3]
```diff
@@ -594,12 +594,14 @@ type migrateTracker struct {
...
 	ns := info.Namespace
 	if len(ns) > 0 {
-		ns = "-n " + ns
+		ns = " -n " + ns
 	}
+	groupResource := info.Mapping.Resource.GroupResource()
+	groupResourceStr := (&groupResource).String()
 	if err != nil {
-		fmt.Fprintf(t.out, "E%s %-10s %s %s/%s: %v\n", timeStampNow(), prefix, ns, info.Mapping.Resource.Resource, info.Name, err)
+		fmt.Fprintf(t.out, "E%s %-10s%s %s/%s: %v\n", timeStampNow(), prefix, ns, groupResourceStr, info.Name, err)
```
Comment (juanvallejo) [inline pkg/oc/cli/admin/migrate/migrator.go]: Is there no feasible way to use our printing stack here? Seem like the whole point of having one was to avoid manually crafting messages like these.

If not feasible to use here, then this is fine. cc @soltysh

File: pkg/oc/cli/admin/migrate/migrator.go (hunk 1 above)
Comment (enj) [reply pkg/oc/cli/admin/migrate/migrator.go]: Yup

File: pkg/oc/cli/admin/migrate/migrator.go (hunk 2 above)
Comment (enj) [reply pkg/oc/cli/admin/migrate/migrator.go]: Should only be used if namespace is non-empty.

File: pkg/oc/cli/admin/migrate/migrator.go (hunk 3 above)
Comment (enj) [reply pkg/oc/cli/admin/migrate/migrator.go]: You are welcome to change it as you see fit, but I want this fixed as-is first.

Comment (soltysh) [reply pkg/oc/cli/admin/migrate/migrator.go]: > If not feasible to use here, then this is fine. cc @soltysh

Sort of yes, but we can generalize all of the use-cases. This particular one is doing more than our printing stack does. So I'm fine with this approach. What we could do is to provide some helpers to getting those group resource string, so that this doesn't have to be replicated over and over again, if anything.

Comment (juanvallejo) [reply pkg/oc/cli/admin/migrate/migrator.go]: > What we could do is to provide some helpers to getting those group resource string, so that this doesn't have to be replicated over and over again, if anything.

+1
